util/util.cpp

```diff
@@ -59,32 +74,56 @@ int getSpecTemp(struct conf::sensorConfig config)
         if (sensorSpecFile)
         {
             sensorSpecFile >> specTemp;
+            sensorSpecFile.close();
         }
-
-        sensorSpecFile.close();
     }
     else if (config.parametersType == "file")
     {
         sensorSpecFile.open(config.parametersPath, std::ios::in);
         if (sensorSpecFile)
         {
             sensorSpecFile >> specTemp;
+            sensorSpecFile.close();
         }
+    }
 
-        sensorSpecFile.close();
```

Comment:
Unfortunately, GitHub doesn't let you see as much surrounding context as Gerrit does, when reviewing patches. The close() was moved to within the block that contains the successful open(). That way, we aren't trying to close a non-open file descriptor, and we're also remembering to close it before returning. There's at least one file descriptor leak bug that was fixed by doing these.

Reply:
Hi Krellan,
Thanks to improvet this repo. It seems fix some bugs that I didn't find before.
About this situation, is it possible that use exception std::fstream::failure to catch error when file open failed?

Reply:
You're welcome. That is something I will look at in the future. I don't think there's a need to use exceptions, the stream can be tested immediately after opening, using std::ios::operator bool(), to detect failure to open, which is already done here. There is something missing, though, an error message if the file could not be opened. This could be added, throttled like the others should be.